libgnucash/engine/gnc-budget.cpp

```diff
@@ -461,6 +475,9 @@ gnc_budget_set_num_periods(GncBudget* budget, guint num_periods)
     qof_instance_set_dirty(&budget->inst);
     gnc_budget_commit_edit(budget);
 
+    for (auto& it : priv->acct_map)
+        it.second.resize (num_periods);
```

Comment:
Resizing a std::vector may discard some `PeriodData` structs. The latter contains an std::string. Can this resize dive into the struct to free the std::string properly?

Reply:
This should be done in the destructor of the struct.

Reply:
>  Can this resize dive into the struct to free the std::string properly?

Yes. The vector contains actual structs as it should, and PeriodData holds the actual strings so removing the elements from the vector destroys all of the struct's contents including the std::string. The std::string manages its character memory so you don't have to do anything.

> This should be done in the destructor of the struct.

It should not. There's nothing to do, the struct holds the string itself so when the PeriodData is destroyed the string goes out of scope and is automatically destroyed. Had the struct held a naked std::string* *then* `~PeriodData` would need to free it, but since C++11 naked pointers are considered [bad practice](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-newdelete), one [should use smart pointers](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-owner), preferably [std::unique_ptr](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-make_unique) instead.